packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -294,11 +295,38 @@ const ProductDetail = () => {
                 }
             ]
 
-            await addItemToBasketMutation.mutateAsync({
+            const res = await addItemToBasketMutation.mutateAsync({
                 parameters: {basketId: basket.basketId},
                 body: productItems
             })
 
+            // the response from addItemToBasketMutation returns an entire basket including newly added bundle product along with other products in the cart
+            // there is no reliable way to filter out the bundle among productItems
+            // Hence, we have to loop over each basket items, and its bundleProductItems to search out the children that still have
+            // masterId and update it with the variantId
+            res.productItems.forEach((productItem) => {
+                productItem.bundledProductItems.forEach((item) => {
```

Comment:
```suggestion
                productItem.bundledProductItems?.forEach((item) => {
```

products that aren't bundles won't have this property